### ISSUE TITLE
:grey_question: API endpoint to trigger a notebook execution :nerd_face: 

### ISSUE THREAD

**Issue by adriens** [Kaggle/kaggle-api#483]
Hi, I wonder is, from a `API` call we could trigger a Notebook execution :grey_question: :crossed_fingers:

**Comment by Philmod** [comment]
Hey @adriens, this endpoint doesn't exist for the moment unfortunately.

**Comment by adriens** [comment]
Any chance to get it :crossed_fingers: :pray:

**Comment by jplotts** [comment]
Hi @adriens - can you clarify the request?  The `kaggle kernels push` command automatically executes the notebook after it's uploaded.  Is this not what you're looking for?

**Comment by adriens** [comment]
Hi, nope. In fact, I would like to be able to **remotely  trigger GPU based notebooks runs from remote places** (like GH Actions), as part of "complex workflows".

**Comment by jplotts** [comment]
@adriens - inside the `kernel-metadata.json` is a flag called `enable_gpu`, so GPU shouldn't be a problem.  I think what you're asking is there an endpoint that will trigger an existing kaggle notebook to run?  If so, there's no API for that directly, but you can use `kaggle kernels push` to run a GPU notebook, as long as you have access to the source and a `kernel-metadata.json`.

Is your request to make this process easier by not requiring the source/metadata?  Or is there some missing functionality?

**Comment by adriens** [comment]
>  I think what you're asking is there an endpoint that will trigger an existing kaggle notebook to run? 

Yup, that's exactly what I would like to play with : it would respond a [`202 Accepted`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202), form a pure `API` perspective :thought_balloon: 

> Is your request to make this process easier by not requiring the source/metadata? Or is there some missing functionality?

My concern would to be able to trigger GPU based notebooks from:

- :octocat:  GitHub Actions
- **⚙️  Any other automation** tool like IFTTT, [Make](https://www.make.com/en), [`n8n`](https://n8n.io/), Power Automate or nay other automation tool
- :robot:  AI autonomous agents (eg while I'm asking them to achieve a mission for me I have packaged as a Langchain tool)
- ...


> but you can use kaggle kernels push to run a GPU notebook, as long as you have access to the source and a kernel-metadata.json.
> 
> Is your request to make this process easier by not requiring the source/metadata? Or is there some missing functionality?
> 

Yup, I was aware of that, it works on my machine but I wanted to be able to run it anytwhere in a more convenient way, which is for me:

- ** :+1:  Ideally no required third party software** (but only installing [`kaggle-api`](https://github.com/Kaggle/kaggle-api) could be :ok:  )
-  :-1: No need to store any additional `metadata.json`
- :-1: Be able to pick a secret from an env variable (eg, I would like to pick it from a given [GH secret's action](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions) or somehting similar)
